beetmoverscript/script.py

```diff
@@ -103,6 +103,7 @@
     product = context.task['payload']['product']
     build_number = context.task['payload']['build_number']
     version = context.task['payload']['version']
+    context.bucket_name = get_bucket_name(context, product)
```

Comment:
How about instead of adding it here, we add it earlier in the stream so that all the types of actions have it?  I'm thinking of something like this, [here](https://github.com/mozilla-releng/beetmoverscript/blob/master/beetmoverscript/script.py#L206).

```
#determine the S3 bucket name from config
context.bucket_name = get_bucket_name(context, context.task['payload']['product'])
```

This was we can spare both this call, which is locally to `push_to_releases` only but also the `upload_to_s3` which is used by `push_to_nightly`.

Optionally, to improve clarity, it might be worth renaming those functions too:
s/`get_task_bucket`/`get_bucket_from_scopes`
s/`get_task_action`/`get_action_from_scopes`

Reply:
`context.task['payload']['product']` doesn't exist outside of push-to-releases. In nightly/candidates, we get the product from the AppName in the balrog manifest. We could change that...

Reply:
Filed https://github.com/mozilla-releng/beetmoverscript/issues/91 to track this.